evm/p2p/lightchain.py

```diff
@@ -265,9 +265,10 @@ def peer_block_height(peer: LESPeer):
         done, pending = await asyncio.wait(
             [self._announcement_queue.get(), wait_for_stop_event()],
             return_when=asyncio.FIRST_COMPLETED)
-        # The async call above returns as soon as one of our 2 coroutines complete, so we know
-        # for sure we'll have one task in the <done> and one task in the <pending> set.
-        pending.pop().cancel()
+        # The asyncio.wait() call above may return both tasks as done, so check that we have
+        # a pending task before attempting to cancel it.
+        if len(pending) > 0:
```

Comment:
Do you think `while pending:` is more legible?

Reply:
`while pending:` or `if pending:`? There would never be more than one pending task here

Reply:
Ok, now that I went and read what `asyncio.wait` does that makes sense.  

Would it be easy at some point in the future for a change to be made to the code above such that `pending` then has more than one *thing* in it at which point this would fail to stop the additional one?  Hence it seems appropriate to write this code in a way that deals with all items in the iterable rather than special casing it for a single item.

Suggestion was more about semantic correctness than the specifics of this functionality.  I'll defer to what you think is best.